### PR TITLE
feat: changed passing token using request HTTP header instead of the request parameter for every API calls

### PR DESCRIPTION
**Pull request by drobnikj**
- [ ] Run integrations test with the new version of the client

### CODE REVIEW

**Comment by drobnikj** [comment]
I just realize that I can probably remove all `this._params()` calls. We probably do not need it in the future.

File: src/base/resource_client.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class ResourceClient extends ApiClient {
         const requestOpts = {
             url: this._url(),
             method: 'GET',
-            params: this._params(options),
+            params: options,
```
Comment (mnmkng) [inline src/base/resource_client.js:33]: We should make a copy of the options where we're passing the options object directly and not recreating or destructuring. This is asking for weird bugs in user code.

**Review by mnmkng — CHANGES_REQUESTED** [review]
Looks good besides that one nitpick 👍

**Review by mtrunkat — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-1.4.0 / 2021/07/29
+1.4.0 / 2021/08/24
```
Comment (B4nan) [inline CHANGELOG.md:1]: next release will be 2.0, this needs rebase

Comment (drobnikj) [reply CHANGELOG.md:1]: We decided to release it as 1.4.0 before the TS rewrite

Comment (B4nan) [reply CHANGELOG.md:1]: master is already v2, so you will need to create a new branch from latest 1.3 and target that instead

**Review by mnmkng — APPROVED** [review]

**Review by B4nan — APPROVED** [review]

**Comment by B4nan** [comment]
cc @vladfrangu this will need to be backported to master once merged

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -5,6 +5,8 @@ on:
   push:
     branches:
       - master
+      # This will allow releasing beta versions from non-master releases.
+      # TODO: We need to release this version with other tags than beta. It can mess up the beta version as it will override it with older code.
```
Comment (B4nan) [inline .github/workflows/release.yml:9]: mmm good catch, but you should have addressed this before merging, as beta releases are automated based on commits to those branches

**Pull request merged**